ecm: 7.0.4 -> 7.0.6

roconnor opened the pull request:
INRIA seems to have moved from gforge to gitlab, so sources have been updated.

I also updated the license which was changed to gpl3 sometime ago.

Fixes #368655.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fee1-dead commented on pkgs/by-name/ec/ecm/package.nix:
```diff
@@ -4,22 +4,29 @@
   fetchurl,
   gmp,
   m4,
+  coreutils,
```
Was this used?

Atemu replied:
No, please remove it.

roconnor replied:
Oops.  At one point I had `${coreutils}/bin/rm`, but I found out I just needed `rm`.  Will fix.

Atemu commented on pkgs/by-name/ec/ecm/package.nix:
```diff
@@ -33,8 +40,8 @@ stdenv.mkDerivation {
     mainProgram = "ecm";
-    license = lib.licenses.gpl2Plus;
-    homepage = "http://ecm.gforge.inria.fr/";
+    license = lib.licenses.gpl3;
```
Please specify gpl3Plus or gpl3Only.

Atemu reviewed (approved):
Please fix these, otherwise intention and diff LGTM and it evidently builds.

roconnor committed:
ecm: 7.0.4 -> 7.0.6

INRIA seems to have moved from gforge to gitlab, so sources have been updated.

I also updated the license which was changed to gpl3 sometime ago.
The commit message says gpl3 or later, but the text of the COPYING does not.
I'm going with gpl3Only as gitlab says when viewing

    https://gitlab.inria.fr/zimmerma/ecm/-/blob/master/COPYING
